### PR TITLE
migrate ReceiveKeyDialog to MaterialDialog to enable dark theme

### LINKED ISSUE #2784: New identity popup dialog wrong theme
Latest HEAD
Using dark theme, when tapping in the conversation on the "changend identiy" bubble, the popup dialog uses the white theme.
when tapping 'verify' from within the details view the dark themed popup appears.

EDIT thinking about it, the detail dialog should not show the verify message, when opening the already verified message. Does not appear after reentering the conversation and opening the message details.

### PR DESCRIPTION
**Pull request by agrajaghh**
This fixes #2784

I used material-dialogs for the new identity dialog (old `ReceiveKeyActivity`) since it's just some text and two buttons. This enables automatic theming. I also removed `MemoryCleaner.clean(masterSecret)`, but it's commented out and will be removed anyway, no?

~~The default text size of the MaterialDialog seems bigger and the spacing between the lines as well. So you need to scroll on small screens to see all the text (buttons are always visible). Should I make the text smaller?~~

### CODE REVIEW

**Comment by agrajaghh** [comment]
rebased, but currently getting this when launching `VerifyIdentityActivity`:
````
    24231-24231/org.thoughtcrime.securesms E/AndroidRuntime﹕ FATAL EXCEPTION: main
    Process: org.thoughtcrime.securesms, PID: 24231
    java.lang.RuntimeException: Unable to resume activity {org.thoughtcrime.securesms/org.thoughtcrime.securesms.VerifyIdentityActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.util.Locale.equals(java.lang.Object)' on a null object reference
            at android.app.ActivityThread.performResumeActivity(ActivityThread.java:2951)
            at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:2982)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2365)
            at android.app.ActivityThread.access$800(ActivityThread.java:144)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1278)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:135)
            at android.app.ActivityThread.main(ActivityThread.java:5221)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.util.Locale.equals(java.lang.Object)' on a null object reference
            at org.thoughtcrime.securesms.util.DynamicLanguage.onResume(DynamicLanguage.java:24)
            at org.thoughtcrime.securesms.KeyScanningActivity.onResume(KeyScanningActivity.java:56)
            at org.thoughtcrime.securesms.VerifyIdentityActivity.onResume(VerifyIdentityActivity.java:73)
            at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1241)
            at android.app.Activity.performResume(Activity.java:6019)
            at android.app.ActivityThread.performResumeActivity(ActivityThread.java:2940)
            at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:2982)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2365)
            at android.app.ActivityThread.access$800(ActivityThread.java:144)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1278)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:135)
            at android.app.ActivityThread.main(ActivityThread.java:5221)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)

````

**Comment by agrajaghh** [comment]
rebased, working now :)

File: src/org/thoughtcrime/securesms/ConversationItem.java [hunk 1]
```diff
@@ -417,17 +416,14 @@ public void onClick(View v) {
...
-    intent.putExtra("thread_id", messageRecord.getThreadId());
-    intent.putExtra("message_id", messageRecord.getId());
-    intent.putExtra("is_bundle", messageRecord.isBundleKeyExchange());
-    intent.putExtra("is_identity_update", messageRecord.isIdentityUpdate());
-    intent.putExtra("is_push", messageRecord.isPush());
-    intent.putExtra("sent", messageRecord.isOutgoing());
-    context.startActivity(intent);
+    NewIdentityKeyDialogBuilder builder;
+    builder = new NewIdentityKeyDialogBuilder(context,
+                                          messageRecord.getIndividualRecipient(),
```
Comment (mcginty) [inline src/org/thoughtcrime/securesms/ConversationItem.java]: ` ` ` ` ` ` ` `

File: src/org/thoughtcrime/securesms/NewIdentityKeyDialogBuilder.java [hunk 2]
```diff
@@ -148,23 +136,22 @@ private IdentityKey getIdentityKey(IncomingPreKeyBundleMessage message)
     }
   }
 
-  private class OkListener implements View.OnClickListener {
+  private class OkCallback extends  MaterialDialog.ButtonCallback {
     @Override
-    public void onClick(View v) {
+    public void onPositive(MaterialDialog dialog) {
       new AsyncTask<Void, Void, Void> () {
```
Comment (mcginty) [inline src/org/thoughtcrime/securesms/NewIdentityKeyDialogBuilder.java]: This seems like a job for `ProgressDialogAsyncTask`!

**Comment by agrajaghh** [comment]
` ` ` ` ` ` ` ` fixed, using `ProgressDialogAsyncTask` now, and switched the `ProgressDialog` in `ProgressDialogAsyncTask` to `MaterialDialog`

I also updated the `MaterialDialog` dependency, there were some bugs fixed, and also the line spacing reduced which shows more text without scrolling for our dialogs with a lot of text...

**Comment by mcginty** [comment]
Do we need both this dialog builder and ConfirmIdentityDialog? They're very similar.

**Comment by agrajaghh** [comment]
hm, I thought I found every AlertDialog, must have missed this one... Will try to combine them.

**Comment by agrajaghh** [comment]
updated, now :small_red_triangle_down:269 lines :blush:

File: src/org/thoughtcrime/securesms/ConfirmIdentityDialogBuilder.java [hunk 3]
```diff
@@ -207,7 +196,6 @@ private VerifySpan(Context context, MasterSecret masterSecret, IdentityKeyMismat
     public void onClick(View widget) {
       Intent intent = new Intent(context, VerifyIdentityActivity.class);
       intent.putExtra("recipient", mismatch.getRecipientId());
-      intent.putExtra("master_secret", masterSecret);
```
Comment (agrajaghh) [inline src/org/thoughtcrime/securesms/ConfirmIdentityDialogBuilder.java]: just to be sure: there aren't any `putExtra("master_secret", masterSecret);` calls after the routing activity change supposed to be in the code if the activity extends `PassphraseRequiredActionBarActivity`, right?

File: src/org/thoughtcrime/securesms/ConversationItem.java [hunk 4]
```diff
@@ -413,20 +414,21 @@ public void onClick(View v) {
     contactPhoto.setVisibility(View.VISIBLE);
   }
 
+  private IdentityKeyMismatch getKeyMismatch(final MessageRecord record) {
```
Comment (agrajaghh) [inline src/org/thoughtcrime/securesms/ConversationItem.java]: I copied this method from https://github.com/WhisperSystems/TextSecure/blob/master/src/org/thoughtcrime/securesms/MessageRecipientListItem.java#L143. Where do you think its best to put it? Just making it `public static` there didn't seem like a good idea...

**Comment by agrajaghh** [comment]
depending on #3072 since all old versions are missing on jCenter..

**Comment by agrajaghh** [comment]
rebased

**Pull request closed without merging**